policy-controller/k8s/status/src/index.rs

```diff
@@ -456,6 +529,9 @@ impl Index {
         // return positive status, otherwise, one of them does not exist
         if backend_refs.iter().all(|backend_ref| match backend_ref {
             routes::BackendReference::Service(service) => self.services.contains_key(service),
+            routes::BackendReference::EgressNetwork(egress_net) => {
+                self.egress_networks.contains_key(egress_net)
+            }
```

Comment:
Is this where we should validate that the EgressNetwork backend must be equal to the parent?

Reply:
Yes, correct. Except for the fact that we now need to compute the backend status per parent because the backend status is dependent on the parent type. Namely, you can have an EgressNetwork parent with a Service backend but not the other way around.